woocommerce/woocommerce-admin
# AdvancedFilters: Create workable defaults for Reports that don't have them

Fixes https://github.com/woocommerce/woocommerce-admin/issues/6529

Attempts to add advanced filters to reports that don't already have them was causing fatal errors when extension devs used the prescribed methods of extending.

This PR add better defaults and a check to see if any advanced filters have been added. If they have, add a regular filter "Advanced filters" so the newly added filters can be accessed.

### Screenshots

<img width="719" alt="Screen Shot 2021-06-16 at 3 59 01 PM" src="https://user-images.githubusercontent.com/1922453/122155825-ddc7d800-cebb-11eb-834e-32330c947007.png">

### Detailed test instructions:
Instructions follow the first few steps of https://developer.woocommerce.com/2020/02/20/extending-wc-admin-reports in order to replicate Filters alongside Advanced Filters.

1. Create a new plugin if you don't already have one.

```
npm run create-wc-extension
cd ../<my-plugin-name>
npm install
npm start
```
2. Visit the Taxes Report and make sure filters work correctly.
3. Add an advanced filter in JS for the Taxes Report, which does not currently have any.

```js
const addShippedDateAdvancedFilter = ( advancedFilters ) => {
	advancedFilters.filters.Shipped = {
		labels: {
			add: __( 'Shipped Date', 'woocommerce-admin' ),
			remove: __( 'Remove Shipped Date filter', 'woocommerce-admin' ),
			rule: __( 'Select Shipped Date', 'woocommerce-admin' ),
			title: __(
				'{{title}}Shipped Date{{/title}} {{rule /}} {{filter /}}',
				'woocommerce-admin'
			),
			filter: __( 'Select Shipped Date', 'woocommerce-admin' ),
		},
		rules: [
			{
				value: 'between',
				label: __( 'Between', 'woocommerce-admin' ),
			},
		],
		input: {
			component: 'Date',
		},
	};

	return advancedFilters;
};

addFilter(
	'woocommerce_admin_taxes_report_advanced_filters',
	'lab-reporting-extension',
	addShippedDateAdvancedFilter
);
```
4. Visit the Taxes Report again and see the regular filters now have an option "Advanced filters"
5. Add an advanced filter and see the url update with parameters of the advanced filter
